extensions/amp-next-page/0.1/next-page-service.js

```diff
@@ -173,6 +174,15 @@ export class NextPageService {
       win.document.title,
       canonicalUrl
     );
+
+    // TODO(wassgha): Establish parity with the shadow doc amp object
+    documentRef.amp = {
```

Comment:
The type definition here does not match the `ShadowDoc` type definition. Title and canonicalUrl are not present in the definition, and resources, which is non-nullable is missing.

I think you should pull this out to a separate line and type it as `ShadowDoc`, Not sure if that would fail type checking, but worth a shot. For example:

```javascript
/** @type {!ShadowDoc} */
const amp = {
   ...
}
```

Reply:
It's not supposed to be 1:1 compatible with `ShadowDoc` for now, that's what the `TODO` is for (probably going to move towards having the first doc be a special type)

Reply:
The types should still match, even if it does not implement the full set. You don't want a case where you are reading a property that may be `undefined` without explicitly handling that case.

This could be a subtype of `ShadowDoc` if you want to track extra fields, but it should at least conform to the contract of that type.

Reply:
Oh nevermind, I see what you mean. It's actually not supposed to be extra fields but rather I forgot to include those fields in the `ShadowDoc` type.